core/src/test/java/test/failedreporter/issue2517/FailedDataProviderWithFactoryReporterSampleTest.java

```diff
@@ -0,0 +1,31 @@
+package test.failedreporter.issue2517;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+public class FailedDataProviderWithFactoryReporterSampleTest {
```

Comment:
Please rename the sample from `...SampleTest` to `...Sample`: it is our convention in order to distinct TestNG tests and test samples for TestNG tests.

Reply:
done